### ISSUE TITLE
child.expect() does not respect timeout when use_poll=True

### ISSUE THREAD

**Issue by ryanpetrello** [pexpect/pexpect#491]
Run the following Python script that _should_ time out after five seconds:

```python
import pexpect
patterns = {
    pexpect.TIMEOUT: None,
    pexpect.EOF: None
}
child = pexpect.spawn('sleep', ['100'], ignore_sighup=True, use_poll=True)
while child.isalive():
    child.expect([], timeout=5, searchwindowsize=100)
```

If you _remove_ the `use_poll=True` argument to `spawn()`, then the `expect()` times out as expected.

related: https://github.com/pexpect/pexpect/pull/474

**Comment by cooperlees** [comment]
Sorry about this. Interesting, I haven't had it come up yet. I'll check how poll takes timeout and find what error I made when I next get time.

**Comment by ryanpetrello** [comment]
Looks like it's stuck on an `os.read()`:

```python
^C
Traceback (most recent call last):
  File "foo.py", line 8, in <module>
    child.expect([], timeout=5, searchwindowsize=1)
  File "/home/ryan/venvs/dev-EKzvPJyz/lib/python2.7/site-packages/pexpect/spawnbase.py", line 341, in expect
    timeout, searchwindowsize, async_)
  File "/home/ryan/venvs/dev-EKzvPJyz/lib/python2.7/site-packages/pexpect/spawnbase.py", line 369, in expect_list
    return exp.expect_loop(timeout)
  File "/home/ryan/venvs/dev-EKzvPJyz/lib/python2.7/site-packages/pexpect/expect.py", line 111, in expect_loop
    incoming = spawn.read_nonblocking(spawn.maxread, timeout)
  File "/home/ryan/venvs/dev-EKzvPJyz/lib/python2.7/site-packages/pexpect/pty_spawn.py", line 485, in read_nonblocking
    return super(spawn, self).read_nonblocking(size)
  File "/home/ryan/venvs/dev-EKzvPJyz/lib/python2.7/site-packages/pexpect/spawnbase.py", line 166, in read_nonblocking
    s = os.read(self.child_fd, size)
KeyboardInterrupt
```

**Comment by takluyver** [comment]
Closed by #492
